### PR TITLE
Merge Sort Implementation for Computer Vision Bootcamp

### PR DESCRIPTION
**Pull request by vigsivan**

### CODE REVIEW

**Review by benjaminwinger — CHANGES_REQUESTED** [review]
Looks good in general, though you need to add the module to the list of modules in the root CMakeLists.txt file. Currently your code isn't actually being compiled and your test isn't being run.

**Comment by vigsivan** [comment]
Hi, I made the changes requested. For the linker issue with the template function, I modified the functions themselves (so that they are no longer template functions) for simplicity's sake.

File: modules/sort/test/test.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  // this is the expected output
+  std::vector<int> soln = {1, 2, 2, 3, 4, 5, 6, 6};
+
+  int max = example.size();
+
+  merge_sort(example, 0, max - 1);
+
+
+  for(int i = 0; i < max; i++){
+    BOOST_TEST(example[i] == soln[i]);
```
Comment (benjaminwinger) [inline modules/sort/test/test.cpp]: The BOOST_TEST macro was actually added in 1.59, however the build server is using 1.58 (I'm in the process of updating it).
Noting that we are planning on using features introduced in boost 1.62, this isn't exactly an issue, so this is fine, but that's why the build server is reporting a failure.

**Comment by benjaminwinger** [comment]
I've added you to Asana, so you can now take a look at the tasks that need to be done. You should probably start with a task marked beginner, as those require less familiarity with the existing code. If you want any suggestions as to what to choose, feel free to contact me on slack. You also should probably let me know before you start working on something just because some of the tasks are a little out of date.

**Pull request closed without merging**